pkg/snapstore/abs_snapstore_test.go

```diff
@@ -7,286 +7,163 @@ package snapstore_test
+func (c *fakeBlockBlobClient) CommitBlockList(ctx context.Context, base64BlockIDs []string, options *blockblob.CommitBlockListOptions) (blockblob.CommitBlockListResponse, error) {
+	keys := []string{}
+	for key := range c.staging {
+		keys = append(keys, key)
 	}
-	w.Body = http.NoBody
-}
+	sort.Strings(keys)
 
-// handleBlobGetOperation on GET request `/testContainer/testObject` responds with a `Get` response.
-func (p *fakePolicy) handleBlobGetOperation(w *http.Response) {
-	key := parseObjectNamefromURL(w.Request.URL)
-	if _, ok := p.objectMap[key]; ok {
-		w.StatusCode = http.StatusOK
-		w.Body = io.NopCloser(bytes.NewReader(*p.objectMap[key]))
-	} else {
-		w.StatusCode = http.StatusNotFound
-		w.Body = http.NoBody
+	contents := []byte{}
+	for _, key := range keys {
+		contents = append(contents, c.staging[key]...)
 	}
-}
 
-// handleDeleteObject on delete request `/testContainer/testObject` responds with a `Delete` response.
-func (p *fakePolicy) handleDeleteObject(w *http.Response) {
-	key := parseObjectNamefromURL(w.Request.URL)
-	if _, ok := p.objectMap[key]; ok {
-		delete(p.objectMap, key)
-		w.StatusCode = http.StatusAccepted
-	} else {
-		w.StatusCode = http.StatusNotFound
-	}
-	w.Body = http.NoBody
+	c.commitFn(&contents)
+	c.staging = make(map[string][]byte)
+
+	return blockblob.CommitBlockListResponse{}, nil
 }
 
-/////////////////////////////////////////////
-// Truncated azblob types for XML encoding //
-/////////////////////////////////////////////
+// StageBlock "uploads" to the "staging" area for the blobs
+func (c *fakeBlockBlobClient) StageBlock(ctx context.Context, base64BlockID string, body io.ReadSeekCloser, options *blockblob.StageBlockOptions) (blockblob.StageBlockResponse, error) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
 
-type blobItem struct {
-	// XMLName is used for marshalling and is subject to removal in a future release.
-	XMLName    xml.Name              `xml:"Blob"`
-	Name       string                `xml:"Name"`
-	Properties azblob.BlobProperties `xml:"Properties"`
-}
+	contents := bytes.NewBuffer([]byte{})
+	if _, err := io.Copy(contents, body); err != nil {
+		return blockblob.StageBlockResponse{}, fmt.Errorf("error while staging the block: %w", err)
+	}
 
-// listBlobsFlatSegmentResponse - An enumeration of blobs
-type listBlobsFlatSegmentResponse struct {
-	XMLName         xml.Name            `xml:"EnumerationResults"`
-	ServiceEndpoint string              `xml:"ServiceEndpoint,attr"`
-	ContainerName   string              `xml:"ContainerName,attr"`
-	Prefix          string              `xml:"Prefix"`
-	Marker          string              `xml:"Marker"`
-	MaxResults      int32               `xml:"MaxResults"`
-	Segment         blobFlatListSegment `xml:"Blobs"`
-	NextMarker      string              `xml:"NextMarker"`
-}
+	decodedBytes, _ := base64.StdEncoding.DecodeString(base64BlockID)
+	decoded := string(decodedBytes)
+	c.staging[decoded] = contents.Bytes()
 
```

Comment:
```suggestion
	contents := bytes.NewBuffer([]byte{})
	if _, err := io.Copy(contents, body); err != nil {
		return blockblob.StageBlockResponse{}, fmt.Errorf("error while staging the block: %w", err)
	}

	decodedBytes, _ := base64.StdEncoding.DecodeString(base64BlockID)
	decoded := string(decodedBytes)
    c.mutex.Lock()
	defer c.mutex.Unlock()
	c.staging[decoded] = contents.Bytes()

```
Since the only race condition can happen is with `c.staging` map, I feel we can move the lock to below so that any goroutines which fail to copy can fail fast and also to let these copy operations & decoding to go on for the other goroutines as those can be quite some time taking steps. Wdyt?